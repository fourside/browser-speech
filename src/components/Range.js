// @flow
import React from 'react';

type Props = {
  name: string,
  min: number,
  max: number,
  step: number,
  value: number,
  onRangeChange: (e: SyntheticEvent<HTMLInputElement>) => void
};

export default class Range extends React.Component<Props> {

  shouldComponentUpdate(nextProps: Props) {
    return this.props.value !== nextProps.value;
  }

  render() {
    const name = this.props.name;
    const min = this.props.min;
    const max = this.props.max;
    const step = this.props.step;
    const value = this.props.value;
    return (
      <div className="form-group">
        <label htmlFor={name}> {name} </label> : {value}
        <input type="range" id={name} min={min} max={max} step={step} value={value} className="form-control"
          onChange={(e) => this.props.onRangeChange(e)}/>
      </div>
    );
  }
}

