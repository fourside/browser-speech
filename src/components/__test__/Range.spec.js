import React from 'react';
import Range from '../Range';
import renderer from 'react-test-renderer';

describe('Range', () => {
  it('should be rendered without crashing.', () => {
    const tree = renderer
      .create(<Range
          name="test" min={0} max={2} step={0.5} value={1.0} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

