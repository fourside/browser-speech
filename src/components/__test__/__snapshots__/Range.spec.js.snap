// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range should be rendered without crashing. 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="test"
  >
     
    test
     
  </label>
   : 
  1
  <input
    className="form-control"
    id="test"
    max={2}
    min={0}
    onChange={[Function]}
    step={0.5}
    type="range"
    value={1}
  />
</div>
`;
